### PR TITLE
Added tool project and fixed directory.build.props

### PR DESCRIPTION
**Pull request by jaceks2106**

### CODE REVIEW

File: sample/Kmd.Logic.Gateway.Automation.Sample/Kmd.Logic.Gateway.Automation.Sample.csproj [hunk 1]
```diff
@@ -32,11 +26,4 @@
       <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
     </None>
   </ItemGroup>
-
-  <ItemGroup>
```
Comment (Araballi) [inline sample/Kmd.Logic.Gateway.Automation.Sample/Kmd.Logic.Gateway.Automation.Sample.csproj:36]: Why this has been removed

File: src/Kmd.Logic.Gateway.Automation.Tool/CommonOptions.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+﻿using CommandLine;
+
+namespace Kmd.Logic.Gateway.Automation.Tool
+{
+    internal class CommonOptions
```
Comment (Araballi) [inline src/Kmd.Logic.Gateway.Automation.Tool/CommonOptions.cs:5]: Can we give different name then common

File: src/Kmd.Logic.Gateway.Automation.Tool/CommonOptions.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    {
+        [Option('t', "token", Required = true, HelpText = "Token used for authentication with Logic.")]
+        public string AuthToken { get; set; }
+
+        [Option('s', "subscriptionId", Required = true, HelpText = "Subscription ID in Logic.")]
+        public string SubscriptionId { get; set; }
+
+        [Option("verbose", HelpText = "Logs everything")]
+        public bool Verbose { get; set; }
```
Comment (Araballi) [inline src/Kmd.Logic.Gateway.Automation.Tool/CommonOptions.cs:14]: provider Id is missing as parameter, you can add this later also :)

File: src/Kmd.Logic.Gateway.Automation/Kmd.Logic.Gateway.Automation.csproj [hunk 4]
```diff
@@ -1,28 +1,13 @@
...
-  <ItemGroup>
-    <PackageReference Update="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="2.9.8">
-      <PrivateAssets>all</PrivateAssets>
-      <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
-    </PackageReference>
-  </ItemGroup>
-
-  <ItemGroup>
-    <Folder Include="Models\" />
-  </ItemGroup>
```
Comment (Araballi) [inline src/Kmd.Logic.Gateway.Automation/Kmd.Logic.Gateway.Automation.csproj:26]: Why this is removed ?

File: sample/Kmd.Logic.Gateway.Automation.Sample/Kmd.Logic.Gateway.Automation.Sample.csproj [hunk 5]
```diff
@@ -32,11 +26,4 @@
...
-    <PackageReference Update="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="2.9.8">
-      <PrivateAssets>all</PrivateAssets>
-      <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
-    </PackageReference>
-  </ItemGroup>
```
Comment (Araballi) [inline sample/Kmd.Logic.Gateway.Automation.Sample/Kmd.Logic.Gateway.Automation.Sample.csproj:41]: Why it has been removed

File: Directory.Build.props [hunk 6]
```diff
@@ -1,34 +1,21 @@
...
-    <VersionPrefix>0.1.0</VersionPrefix>
-    <Authors>KMD</Authors>
-    <PackageTags>kmd;logic;Gateway</PackageTags>
-    <PackageIconUrl>https://console.kmdlogic.io/favicon-32x32.png</PackageIconUrl>
-    <PackageProjectUrl>https://github.com/kmdlogic/kmd-logic-gateway-automation</PackageProjectUrl>
-    <PackageLicenseExpression>MIT</PackageLicenseExpression>
-    <RepositoryUrl>https://github.com/kmdlogic/kmd-logic-gateway-automation</RepositoryUrl>
-    <RepositoryType>git</RepositoryType>
-    <GenerateAssemblyVersionAttribute>false</GenerateAssemblyVersionAttribute>
-  </PropertyGroup>
```
Comment (Araballi) [inline Directory.Build.props:21]: Why this has been removed

File: sample/Kmd.Logic.Gateway.Automation.Sample/Kmd.Logic.Gateway.Automation.Sample.csproj (hunk 1 above)
Comment (jaceks2106) [reply sample/Kmd.Logic.Gateway.Automation.Sample/Kmd.Logic.Gateway.Automation.Sample.csproj:36]: Why do we need this? It is just sample. This description should be on the dotnet tool nuget package.

File: sample/Kmd.Logic.Gateway.Automation.Sample/Kmd.Logic.Gateway.Automation.Sample.csproj (hunk 5 above)
Comment (jaceks2106) [reply sample/Kmd.Logic.Gateway.Automation.Sample/Kmd.Logic.Gateway.Automation.Sample.csproj:41]: It's already on Directory.Build.props.

File: src/Kmd.Logic.Gateway.Automation.Tool/CommonOptions.cs (hunk 3 above)
Comment (jaceks2106) [reply src/Kmd.Logic.Gateway.Automation.Tool/CommonOptions.cs:14]: Yeah. Here is just a skeleton.

File: Directory.Build.props (hunk 6 above)
Comment (jaceks2106) [reply Directory.Build.props:21]: As far as I know, we only expose dotnet tool as NuGet package, not library or even sample. So IMO we do not need this.

Comment (Araballi) [reply Directory.Build.props:21]: We expose library NuGet also, we give more option to consumer, he can use automation NuGet or Dotnet tool.

File: src/Kmd.Logic.Gateway.Automation.Tool/CommonOptions.cs (hunk 2 above)
Comment (rosieks) [reply src/Kmd.Logic.Gateway.Automation.Tool/CommonOptions.cs:5]: I agree with @Araballi - what about `GatewayCommandOptions`?

Comment (jaceks2106) [reply src/Kmd.Logic.Gateway.Automation.Tool/CommonOptions.cs:5]: My suggestion is to name it just `BaseCommandOptions`. We already know that it is Gateway tool.

File: Directory.Build.props (hunk 6 above)
Comment (jaceks2106) [reply Directory.Build.props:21]: Ooooh, I didn't know. So for sure it must be included.

File: src/Kmd.Logic.Gateway.Automation.Tool/CommonOptions.cs (hunk 2 above)
Comment (jaceks2106) [reply src/Kmd.Logic.Gateway.Automation.Tool/CommonOptions.cs]: Check now.

Comment (mlatoszek) [reply src/Kmd.Logic.Gateway.Automation.Tool/CommonOptions.cs]: LGTM

**Review by mlatoszek — APPROVED** [review]
LGTM but do you need to merge it? It's just a skeleton for dotnet tool and it does not provide value... or maybe you need this to configure dotnet tool release?

**Comment by jaceks2106** [comment]
> LGTM but do you need to merge it? It's just a skeleton for dotnet tool and it does not provide value... or maybe you need this to configure dotnet tool release?

Start small :)

File: src/Kmd.Logic.Gateway.Automation.Tool/CommonOptions.cs [hunk 7]
```diff
@@ -0,0 +1,16 @@
+﻿using CommandLine;
+
+namespace Kmd.Logic.Gateway.Automation.Tool
+{
+    internal class CommonOptions
+    {
+        [Option('t', "token", Required = true, HelpText = "Token used for authentication with Logic.")]
+        public string AuthToken { get; set; }
```
[the remark this answers is not in the document]
Comment (jaceks2106) [reply src/Kmd.Logic.Gateway.Automation.Tool/CommonOptions.cs]: Well noticed. It's already mentioned in our User Story, that passing JWT token is not legit. Client credentials will be applied in the library NuGet package, and then utilized by CLI Tool.

**Pull request merged**